.github/workflows/examples-components.yml

```diff
@@ -77,30 +72,19 @@ jobs:
           # Once a new version of wash is out we can use prebuilt versions again (>= 35.0)
           - current
     steps:
-      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
-        if: ${{ matrix.wash-version == 'current' }}
-      - run: rustup show
-      - uses: Swatinem/rust-cache@82a92a6e8fbeee089604da2575dc567ae9ddeaab
-        with:
-          shared-key: 'ubuntu-22.04-shared-cache'
-      - name: install wash (previous version)
-        if: ${{ matrix.wash-version != 'current' }}
-        uses: taiki-e/install-action@2c3c8dfabf5933531aca0613c6b0c173fa51745e
-        with:
-          tool: wash-cli@${{ matrix.wash-version }}
-      - name: build wash (current)
-        if: ${{ matrix.wash-version == 'current' }}
-        run: cargo build -p wash-cli --release
-      - uses: actions/upload-artifact@6f51ac03b9356f520e9adb1b1b7802705f340c2b
-        if: ${{ matrix.wash-version == 'current' }}
-        with:
-          name: wash-${{ matrix.wash-version }}
-          path: ./target/release/wash
-      - uses: actions/upload-artifact@6f51ac03b9356f520e9adb1b1b7802705f340c2b
-        if: ${{ matrix.wash-version != 'current' }}
-        with:
-          name: wash-${{ matrix.wash-version }}
-          path: ~/.cargo/bin/wash
+    - name: install wash (${{ matrix.wash-version }})
```

Comment:
I'm not sure `${{ matrix.<variable> }}` gets interpolated into the output, afaik it shows up literally as ${{ matrix.... }}

Reply:
Yeah I had to experiment with this a bit. It does show the expression `${{ ... }}` as a string, but only if the matrix isn't evaluated for some reason like an `if: ` condition. We're using in the wasmcloud workflow too:
https://github.com/wasmCloud/wasmCloud/blob/a24e07a03d3b499d8a03b896d93c2c1eb10bf80f/.github/workflows/wasmcloud.yml#L154

 Here's what it shows up as:
<img width="270" alt="Screenshot 2024-12-27 at 5 48 59 PM" src="https://github.com/user-attachments/assets/2c2bfaef-0645-44dc-86f7-5082ec90b0c7" />

This is also nice because you can control what info shows in the title for matrix jobs since it actually ditches the matrix values in the parenthesis from the name.

Reply:
Took it a step further and updated the component/provider build job names:

https://github.com/wasmCloud/wasmCloud/blob/d23683e9859621ab5fa9df9e20686f7db5318f24/.github/workflows/examples-components.yml#L91

<img width="762" alt="Screenshot 2024-12-27 at 6 54 42 PM" src="https://github.com/user-attachments/assets/79cae38e-8c2f-4163-8adb-153b1a784d0e" />